app/uk/gov/hmrc/gform/eval/ExpressionResult.scala

```diff
@@ -101,10 +106,16 @@ sealed trait ExpressionResult extends Product with Serializable {
     case t: OptionResult =>
       er.fold[Boolean](_ => false)(_ => false)(_ => false)(n => t.contains(n.value))(_ => false)(_ => false)(_ =>
         false
-      )(_ => false)(_ => false)
+      )(_ => false)(_ => false)(_ => false)
     case t: DateResult    => false
     case t: PeriodResult  => false
     case t: AddressResult => false
+    case t: ListResult =>
```

Comment:
Can't this be simply 
```
t.list.exists(_.contains(er))
```
So the `exists` semantics is clearly visible?

Reply:
This doesnt seem to work, as contains for StringResult, NumberResult etc seems to default to false

Reply:
`contains` operator is used only in conjunction with choice/revealingChoice components, so no `StringResult` or `NumberResult` should be considered and thus false is correct behaviour.
IOW `${surname contains 'Smith'}` is invalid expression if surname is `type: text`.

Reply:
Yes, for that reason when 

```t.list.exists(_.contains(er))```

evaluates and runs contains on StringResult, it returns false and fails.